mesonbuild/mparser.py

```diff
@@ -203,9 +204,16 @@ def lex(self, filename: str) -> T.Generator[Token, None, None]:
                             value = ESCAPE_SEQUENCE_SINGLE_RE.sub(decode_match, value)
                         except MesonUnicodeDecodeError as err:
                             raise MesonException(f"Failed to parse escape sequence: '{err.match}' in string:\n  {match_text}")
-                    elif tid == 'multiline_string':
-                        tid = 'string'
-                        value = match_text[3:-3]
+                    elif tid in {'multiline_string', 'multiline_fstring'}:
+                        # for multiline strings, parse the value out of the
+                        # quotes, update the line numbers, then funnel into
+                        # the normal string/fstring logic
+                        prefix, tid = {
+                            'multiline_string': ("'''", 'string'),
+                            'multiline_fstring': ("f'''", 'fstring'),
+                        }[tid]
+                        postfix = "'''"
+                        value = match_text[len(prefix):-len(postfix)]
```

Comment:
Shouldn't we just hardcode postfix as 3, and hardcode prefix as either 3 or 4?

Reply:
I'm fine going either way on this. I was trying to make it a little clearer to read, but I can change it to be hardcoded.

Reply:
To be clear, I don't have an issue with the dictionary getter, but `len("f'''")` can just be `4` inside the dictionary.